ykcompile/src/lib.rs

```diff
@@ -1826,4 +1829,29 @@ mod tests {
         assert_eq!(argv.len(), 1);
         assert_eq!(argv[0], 3);
     }
+
+    #[test]
+    fn test_projection_chain() {
+        #[derive(Debug)]
+        struct S {
+            x: usize,
+            y: usize,
+        }
+        let s = S { x: 5, y: 6 };
+        let t = (1usize, 2u8, 3usize);
+        let mut inputs = trace_inputs((t, 0u8, s, 0usize));
+        let th = start_tracing(Some(TracingKind::HardwareTracing));
+        inputs.1 = (inputs.0).1;
+        inputs.3 = inputs.2.y;
+        let sir_trace = th.stop_tracing().unwrap();
+        let tir_trace = TirTrace::new(&*SIR, &*sir_trace).unwrap();
+        let ct = TraceCompiler::<&((usize, u8, usize), u8, S, usize)>::compile(tir_trace);
+
+        let t2 = (1usize, 2u8, 3usize);
+        let s2 = S { x: 5, y: 6 };
+        let mut args = (t2, 0u8, s2, 0usize);
+        ct.execute(&mut args);
+        assert_eq!(args.1, 2);
```

Comment:
We can make this test stronger by checking the whole inputs tuple is the right shape. That way we'd also check the read-only parts are intact.

Reply:
Good point.